.circleci/pyscripts/process_generate.py

```diff
@@ -0,0 +1,220 @@
+# /usr/bin/env python3
+import os
+import re
+import json
+from library import libgit
+from library import libfile
+
+path_to_build_components = os.path.join("/tmp", "build-triggers.json")
+path_to_build_trigger_override = os.path.join(".circleci", "build-triggers.override.json")
+
+output_path = os.environ.get('OUTPUT_PATH')
+head = os.environ.get('CIRCLE_SHA1')
+base_revision = os.environ.get('BASE_REVISION')
+branch_name = os.environ.get('CIRCLE_BRANCH')
+
+libgit = libgit.libgit("stdout")
+
+libfile = libfile.libfile()
+
+
+libgit.switchBranch(base_revision)
+libgit.switchBranch(head)
+
+base = libgit.commonAncestor(base_revision, head)
+
+print("branch_name", branch_name)
+print("output_path", output_path)
+print("head", head)
+print("base_revision", base_revision)
+print("base", base)
+
+if head == base:
```

Comment:
@mershad-manesh mentioned that this code was existing, and it is, but I missed the original when I was looking at the diff. Can you put the comments back in this section? They seem quite useful.

```
-        # If building on the same branch as BASE_REVISION, we will get the
-        # current commit as merge base. In that case try to go back to the
-        # first parent, i.e. the last state of this branch before the
-        # merge, and use that as the base.
-        base = subprocess.run(
-            ['git', 'rev-parse', 'HEAD~1'], # FIXME this breaks on the first commit, fallback to something
-            check=True,
-            capture_output=True
-        ).stdout.decode('utf-8').strip()
-    except:
-        # This can fail if this is the first commit of the repo, so that
-        # HEAD~1 actually doesn't resolve. In this case we can compare
-        # against this magic SHA below, which is the empty tree. The diff
-        # to that is just the first commit as patch.
-        base = '4b825dc642cb6eb9a060e54bf8d69288fbee4904'
```